api/src/test/java/org/openmrs/module/idgen/SequentialIdentifierGeneratorTest.java

```diff
@@ -73,4 +94,49 @@ public void shouldSetNextSequenceValueToNegative() throws Exception {
 		generator.setMaxLength(13);
  		assertThat(generator.getNextSequenceValue(), is(-1l));
 	}
+	
+	@Test
+	public void getIdentifierForSeed_shouldGenerateLocationPrefixedIdFromLocationBasedPrefixProvider() {
+		SequentialIdentifierGenerator generator = new SequentialIdentifierGenerator();
+		generator.setBaseCharacterSet("0123456789");
+		generator.setFirstIdentifierBase("000");
+		generator.setName("Location Prefixed Sequential Identifier Source");
+		generator.setPrefixProviderBean(LocationBasedPrefixProvider.class.getSimpleName());
+		
+		UserContext userContext = mock(UserContext.class);
+		when(userContext.getLocation()).thenReturn(createLocationTree());
+		when(Context.getUserContext()).thenReturn(userContext);
+		when(Context.getRegisteredComponent("LocationBasedPrefixProvider", PrefixProvider.class)).thenReturn(new LocationBasedPrefixProvider());
+		
+		assertThat(generator.getIdentifierForSeed(1L), is("AFDEL-001"));
+	}
+	
+	private Location createLocationTree() {
+		LocationAttributeType prefixAttrType = new LocationAttributeType();
+		prefixAttrType.setName(IdgenConstants.PREFIX_LOCATION_ATTRIBUTE_TYPE);
+		prefixAttrType.setMinOccurs(0);
+		prefixAttrType.setMaxOccurs(1);
+		prefixAttrType.setDatatypeClassname("org.openmrs.customdatatype.datatype.FreeTextDatatype");
+		
+		Location mainReg = new Location();
+		mainReg.setName("Main Registration");
+		
+		Location kch = new Location();
+		kch.setName("Kaboul Central Hospital");
+		kch.addChildLocation(mainReg);
+		
+		LocationAttribute kchPrefixAtt = new LocationAttribute();
+		kchPrefixAtt.setAttributeType(prefixAttrType);
+		kchPrefixAtt.setValue("AFDEL-");
+		kch.addAttribute(kchPrefixAtt);
+		
+		Location ks = new Location();
+		ks.setName("Kaboul Subdelegation");
+		ks.addChildLocation(kch);
+		
+		Location af = new Location();
+		af.setName("Afghanistan Delegation");
+		af.addChildLocation(ks);
+		return mainReg;
```

Comment:
Could you not reuse the same test tree?
If **not**, please give those generic names:
- Registration Hall
- Referral Hospital
- Health District Office
- Health Regional Office